website/docs/concepts/router.mdx

```diff
@@ -44,8 +44,9 @@ current URL and passes it to the `render` callback. The callback then decides wh
 matched, the router navigates to the path with `not_found` attribute. If no route is specified, nothing is rendered, and
 a message is logged to console stating that no route was matched.
 
-Finally, you need to register one of the Router context provider components like `<BrowserRouter />`.
-It provides location information and navigator to its children.
+The router components, e.g. `<Link />` and `<Switch />`, must be used in the [context](concepts/contexts.mdx)
```

Comment:
I'm not sure if I like the wording

> must be used in the context of router provider 

There's no "router provider" (there is, but it's not exposed). The terms "provider" and "context" are implementation details.

How about something like:

> **must** be children of a Router component (e.g. `<BrowserRouter />`).

If we want to specify that Router is a context, this line can also be added:

> This component registers a context, which can only be accessed by it's children.

Maybe we should also mention that while it is possible to nest router components, it is not recommended as it can lead to bugs that show up in weird ways and can be hard to debug.

Reply:
I'd prefer the "context" wording over "children", because if `<A>` renders `<BrowserRouter><B /></BrowserRouter>` and `<B>` renders some `<Link />` then it might not be clear that it's a child in this sense?

Reply:
Originally I opened up this PR to address the fact that it wasn't obvious that `<Link />` (and others) need to be children/ in a context of a Router. I'm ok with either of these, because they both clarify that.

What about

> can only be used in the component tree of a Router component (e.g. `<BrowserRouter />`)

Reply:
Okay, how about combining the proposed phrasings into one:

> Most of `yew-router`'s components, in particular `<Link />` and `<Switch />`, must be (grand-)children of one of the Router components (e.g. `<BrowserRouter />`). You usually only need a single Router in your app, most often rendered immediately by your most top-level <App /> component. The Router registers a [context](concepts/contexts.mdx), which is needed for Links and Switches to function. An example is shown below.

I think this covers all the bases nicely

Reply:
@hamza1311 any thoughts on this? I was waiting to see if you where in agreement with this or not. Personally, I think it's a reasonable solution.

Reply:
Fully agree with @WorldSEnder here. Let's change the wording to what they suggested.